### PR TITLE
Story block: Add more media options to choose from in the editor

### PR DESCRIPTION
**Pull request by Tug**
#### Changes proposed in this Pull Request:
This change allows for selecting media in the story block from the same media sources as the gallery block and other visual blocks:

image

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
- Apply this patch on a jetpack site connected to wordpress.com
- Navigate to a post containing a story or create a new story
- Click on Select Media
- Notice Google Photos and Pexels are available and that we can mix media from different sources when editing a story

### CODE REVIEW

**Review by aforcier — CHANGES_REQUESTED** [review]
I gave this a try, and the first time media are added, it works great! However I ran into some issues when trying to change media several times in one session:

1. Start by adding an empty story block
2. Add media from any source
3. Tap on 'select media' again, and select at least one image from either Google Photos or Pexels
4. The new images are appended correctly, but the previous images become blank
5. Repeat step 3
6. Notice that the new selection was completely ignored

At step 4 I inspected the block, and it looks like this:

```
<!-- wp:jetpack/story {"mediaFiles":[{},{},{"alt":"lake and mountain","id":296,"type":"image","caption":"Photo by James Wheeler on \u003ca href=\u0022https://www.pexels.com/photo/lake-and-mountain-417074/\u0022 rel=\u0022nofollow\u0022\u003ePexels.com\u003c/a\u003e","url":"https://website.com/wp-content/uploads/2021/04/pexels-photo-417074.jpeg"}]} -->
<div class="wp-block-jetpack-story wp-story"></div>
<!-- /wp:jetpack/story -->
```

Interestingly, in the gallery block, once media has been added (from any source), it looks like the options for Google Photos and Pexels are removed, and you can only select media from your media gallery. Perhaps we should do the same?

File: projects/plugins/jetpack/changelog/story-block-add-more-media-options [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: other
+
+Story block: Add more media options to choose from in the editor
```
[the remark this answers is not in the document]
Comment (Tug) [reply projects/plugins/jetpack/changelog/story-block-add-more-media-options:4]: Type `other` is ignored as well. I can add the `Comment` prefix as well 👍

**Comment by Tug** [comment]
@aforcier Thanks for finding and reporting this bug, I should have noticed 😞 . It's fixed now. I'll give it a more thorough round of testing before putting it back for review (need to test on WPCOM for instance)

File: projects/plugins/jetpack/changelog/story-block-add-more-media-options (hunk 1 above)
[the remark this answers is not in the document]
Comment (aforcier) [reply projects/plugins/jetpack/changelog/story-block-add-more-media-options]: Oh interesting, I didn't know that - is that written down somewhere? I didn't see it in [the instructions](https://github.com/Automattic/jetpack/blob/master/docs/monorepo.md#using-the-jetpack-changelogger), maybe they should be updated?

**Comment by aforcier** [comment]
Hey @Tug , I gave this another try, on both Jetpack and WP.com.

Things are working smoothly and the bug seems to be fixed 🎉 I did notice though that the edit button is now text (it used to be a pencil icon before this branch), that seems a bit out of place?

Screen Shot 2021-05-01 at 9 19 24 AM

**Comment by Tug** [comment]
> I did notice though that the edit button is now text

I think that's the default media upload button on WPCOM. On self-hosted + jetpack site, it should still be a pencil icon

**Comment by aforcier** [comment]
>I think that's the default media upload button on WPCOM (unrelated to this change). On self-hosted + jetpack site, it should still be a pencil icon

It looks like that may not be the case @Tug - those screenshots were from a self-hosted site. I checked again, this is a WPcom site before and after this branch:

edit-icon-wpcom-beforeafter

This is a self-hosted site:

edit-icon-jetpack-beforeafter

Any idea what's going on?

**Comment by Tug** [comment]
So it wasn't actually related to WPCOM vs self-hosted, but kind of: 
Jetpack is actually replacing this button with a custom one when 
- we have external services available for media upload (so google photos and pexels), it usually happens once you've connected your account with WordPress.com.
- it's not a gallery (we cannot use galleries as those are only for images)
The logic for this is defined [here](https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/extensions/shared/external-media/index.js#L57).

This is so that the menu with the different sources to choose from can be injected.

We could either leave it as it is or try to update this component to try to re-use the `render` props of the original `MediaUpload` component.

@obenland what do you think? Can it be done without impacting other jetpack blocks?

**Review by aforcier — APPROVED** [review]
Per our discussion, let's not block this PR since keeping the pencil icon is not a straightforward change/involves other components beside the story block. We'll open an issue and look into that separately.

:shipit:

**Comment by Tug** [comment]
Issue reported in https://github.com/Automattic/jetpack/issues/19742

**Review by jeherve — APPROVED** [review]

**Comment by Tug** [comment]
Deployed on WPCOM with Changeset ID 225384

**Pull request merged**